kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net/GwtNetworkServiceImpl.java

```diff
@@ -353,16 +353,23 @@ private static List<GwtNetInterfaceConfig> privateFindNetInterfaceConfigurations
                                 gwtWifiConfig.setDriver(wifiConfig.getDriver());
 
                                 // security
-                                if (wifiConfig.getSecurity() == WifiSecurity.SECURITY_WPA) {
-                                    gwtWifiConfig.setSecurity(GwtWifiSecurity.netWifiSecurityWPA.name());
-                                } else if (wifiConfig.getSecurity() == WifiSecurity.SECURITY_WPA2) {
-                                    gwtWifiConfig.setSecurity(GwtWifiSecurity.netWifiSecurityWPA2.name());
-                                } else if (wifiConfig.getSecurity() == WifiSecurity.SECURITY_WPA_WPA2) {
-                                    gwtWifiConfig.setSecurity(GwtWifiSecurity.netWifiSecurityWPA_WPA2.name());
-                                } else if (wifiConfig.getSecurity() == WifiSecurity.SECURITY_WEP) {
-                                    gwtWifiConfig.setSecurity(GwtWifiSecurity.netWifiSecurityWEP.name());
-                                } else {
-                                    gwtWifiConfig.setSecurity(GwtWifiSecurity.netWifiSecurityNONE.name());
+                                switch(wifiConfig.getSecurity()){
+                                    case SECURITY_WPA:
+                                        gwtWifiConfig.setSecurity(GwtWifiSecurity.netWifiSecurityWPA.name());
+                                        break;
+                                    case SECURITY_WPA2:
+                                        gwtWifiConfig.setSecurity(GwtWifiSecurity.netWifiSecurityWPA2.name());
+                                        break;
+                                    case SECURITY_WPA_WPA2:
+                                        gwtWifiConfig.setSecurity(GwtWifiSecurity.netWifiSecurityWPA_WPA2.name());
+                                        break;
+                                    case SECURITY_WEP:
+                                        gwtWifiConfig.setSecurity(GwtWifiSecurity.netWifiSecurityWEP.name());
+                                        break;
+                                    case SECURITY_NONE:
+                                    default:
```

Comment:
If, in the future, we add a new type to the enum then it is not correct to default to `NONE`. Either:
- add an enum value to `GwtWifiSecurity` for "unsupported" and handle it; or
- restrict the imported `kura.net.wifi` to `[2.4,2.5)`.

I'd prefer the second one.